### ISSUE TITLE
SC2016 false positive

### ISSUE THREAD

**Issue by cornfeedhobo** [koalaman/shellcheck#2168]
#### For bugs
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```plain
        local msg="${2:-Command '$1' does not exist!}"
                                ^--^ SC2016: Expressions don't expand in single quotes, use double quotes for that.
```

However, works as expected:
```plain
Command 'fake-command' does not exist!
```

**Comment by MichaIng** [comment]
I just found the same with this line:
```
echo "curl -sSfL${USERNAME:+ -u '$USERNAME:$PASSWORD'} '$url' 2>&1 > /dev/null | logger -t dietpi-ddns -p 3" >> /var/lib/dietpi/dietpi-ddns/update.sh
```

Generally inside double quotes outside of command substitutions, single quotes have no effect on variable expansion, which includes all sorts of shell string manipulation in `${var...}`.

As fast as you skip the double quotes, which is usually perfectly fine for variable assignment (as word-splitting and globbing is not done when assigning variables or command substitutions to variables), the variable indeed is not expanded:
```
# test=success
# string=${empty:-ab '$test' cd}
# echo "$string"
ab $test c
# string="${empty:-ab '$test' cd}"
# echo "$string"
ab 'success' cd
```

**Comment by kevinoid** [comment]
This looks like a duplicate of #501?

**Comment by MichaIng** [comment]
It is, indeed.

**Comment by cornfeedhobo** [comment]
Looks like it is. Any idea when #501 will get attention?
